src/modules/pos/validator/components/StakeForm/StakeForm.js

```diff
@@ -154,10 +169,44 @@ const StakeForm = ({ t, stakes, account, isStakingTxPending, nextStep, history,
   );
   const showEmptyState = !changedStakes.length || isStakingTxPending;
 
-  const onConfirm = (formProps, transactionJSON, selectedPriority, fees) => {
-    if (!showEmptyState) {
+  const onConfirm = async (formProps, transactionJSON, selectedPriority, fees) => {
+    if (!showEmptyState && moduleCommandSchemas && !isLoading) {
+      const moduleCommand = joinModuleAndCommand(transactionJSON);
+      const paramsSchema = moduleCommandSchemas[moduleCommand];
+      const transaction = fromTransactionJSON(transactionJSON, paramsSchema);
+
+      setIsLoading(true);
+      const [error, dryRunResult] = await to(
+        dryRun({
+          skipVerify: true,
+          paramsSchema,
+          transaction,
+          serviceUrl: mainChainNetwork.serviceUrl,
+        })
+      );
+
+      setIsLoading(true);
+      setDryrunError(error);
+      const rewards = dryRunResult.data.events.reduce((result, event) => {
+        const { name, data, module } = event;
+
+        if (
+          name === 'rewardsAssigned' &&
+          module === 'pos' &&
+          data.stakerAddress === currentAccount.metadata.address
+        ) {
+          return {
+            ...result,
+            [data.validatorAddress]: data,
+            total: BigInt(result.total || 0) + BigInt(data.amount),
```

Comment:
whats this total? we only need `data.amount` 🤔

Reply:
As you can see, the data structure tells us that we should only care about amount https://github.com/LiskHQ/lisk-sdk/blob/98c3d9c588b56845e46eb8a209985a92bf478c67/framework/src/modules/pos/events/rewards_assigned.ts#L18, maybe result property is related to success of failure i guess

Reply:
the reason for total is because we are trying to normalise the response from dry run and total would aggregate all rewards as we normalise the response. This means we don't need to start aggregating total on the summary page with something like this. `Object.values(rewards).reduce((result, { amount}) => result + amount, 0) ` instead we simply just do `rewards.total`